app/tests/test_github_oauth.py

```diff
@@ -0,0 +1,52 @@
+import json
+
+from django.contrib.auth.models import User as DjangoUser
+from django.utils import timezone
+from oauth2_provider.models import AccessToken, Application
+from oauth2_provider.settings import oauth2_settings
+from rest_framework.test import APITestCase
+
+
+class TestAPICallIsValid(APITestCase):
+
+    def setUp(self):
+        self.dev_user = DjangoUser.objects.create(
+            username='test user',
+            email='test@test.com',
+            password='testpassword'
+        )
+
+        self.oauth_test_user = DjangoUser.objects.create_user(
+            username='oauthuser',
+            email='test@email.com',
+            password='123456'
+        )
+
+        self.github_application = Application.objects.create(
+            name="Test application",
+            redirect_uris="http://example.com",
+            user=self.dev_user,
+            client_type=Application.CLIENT_PUBLIC,
+            authorization_grant_type=Application.GRANT_AUTHORIZATION_CODE
+        )
+        self.github_application.save()
+
+        oauth2_settings._SCOPES = ['read', 'write']
+
+        self.access_token = AccessToken.objects.create(
+            user=self.oauth_test_user,
+            scope='read write',
+            expires=timezone.now() + timezone.timedelta(days=1),
+            token='1234567890',
+            application=self.github_application
+        )
+        self.access_token.save()
+
+        self.auth = 'Bearer {0}'.format(self.access_token.token)
+
+    def test_valid_token_registers_new_user_or_validates_existing_user(self):
+
+        response = self.client.get('/github_auth/', HTTP_AUTHORIZATION=self.auth)
+        content = json.loads(response.content.decode('utf-8'))
+        print(content)
+        # self.assertEqual(response.status_code, 200)
```

Comment:
maybe remove comment and print statement?

Reply:
> I find the naming of the User model(DjangoUser, DefaultUser) confusing. Asides that and Kosy's comments, I think it's good to go. Please remember to rebase your commits before you merge.

I added the the oauth test to this branch too, please review first, it doesn't work very well. Explained in the PR description.

Reply:
Yes, but I don't think they were both used in the same place so I don't think calling them `User` affects anything. Maybe just naming them better would work though.

Reply:
Lool I'll think about better names.

Reply:
> maybe remove comment and print statement?

Oh this doesn't work yet, just saw that it's the test file. I'm not merging just yet anyway.

Reply:
never mind, removed this file. Will use a new branch for test